lib/src/c_string.c

```diff
@@ -315,80 +319,82 @@ size_t my_str_find(const my_str_t *str, const my_str_t *tofind, size_t from) {
         return -1;
     return 0;
 }
 
-// TODO: test my_str_cmp_cstr
 int my_str_cmp_cstr(const my_str_t *str1, const char *cstr2) {
-    // compare string with cstring
-    // behavior is the same as in strcmp
-    // return 0 if there are equal, -1 if first is less, 1 if second is less
-    size_t str2_len = char_arr_len(cstr2);
-    if (str1->size_m > str2_len)
+    // This function return values that are as follows:
+    // Return value < 0 if str1 is alphabetical less than str2.
+    // Return value > 0 if str2 is alphabetical less than str1.
+    // Return value = 0 if str1 is equal to str2.
+    size_t i = 0;
+    size_t cstr2_len = char_arr_len(cstr2);
+    while (i < str1->size_m && i < cstr2_len) {
+        if (str1->data[i] > cstr2[i])
+            return 1;
+        else if (str1->data[i] < cstr2[i])
+            return -1;
+        i++;
+    }
+    if ((*str1).size_m > cstr2_len)
         return 1;
-    else if (str1->size_m < str2_len)
+    else if (str1->size_m < cstr2_len)
         return -1;
-    else {
-        for (int j = 0; j < str1->size_m; ++j) {
-            if (str1->data[j] != cstr2[j])
-                return -1;
-        }
-        return 0;
-    }
+    return 0;
 }
 
-// TODO: test my_str_find_c
 size_t my_str_find_c(const my_str_t *str, char tofind, size_t from) {
     // find first symbol in the string
     // from - place from where start searching
     // if from greater then size of string, we cant find char
     // return it`s index or (size_t) (-1) if there no `tofind` chars there.
     if (from > str->size_m)
         return -1;
-    if (from < 0)
-        from = 0;
+
     for (size_t j = from; j < str->size_m; ++j) {
         if (str->data[j] == tofind)
             return j;
     }
     return (size_t) (-1);
 }
 
-// TODO: test my_str_find_if
 size_t my_str_find_if(const my_str_t *str, int (*predicat)(int)) {
-    // find symbol in the string for whish function returns true
-    // return it`s index or (size_t) (-1) if there no such character
-    for (int j = 0; j < str->size_m; ++j) {
-        if (predicat(str->data[j]))
+    // Find the first occurrence of a char that satisfy the 'predicat' function
+    // If there is not such char in the string returns -1
+    for (size_t j = 0; j < str->size_m; ++j) {
+        if (predicat((int) str->data[j]))
             return j;
     }
     return (size_t) (-1);
 }
 
 int my_str_read_file(my_str_t *str, FILE *file) {
-    // read string from file
-    // it reads all file
-    // return 0 if everything ok, negative numbers otherwise
+    // read the file to the EOF
+    // return -1 if can not add a character from the file to the end of the string
+    // return 0 if everything is Ok
     return my_str_read_file_delim(str, file, EOF);
 }
 
 int my_str_read(my_str_t *str) {
-    //analog of my_str_read_file but from stdin
+    // read the input from the standard input till the EOF
+    // return -1 if can not add a character from the file to the end of the string
```

Comment:
I'm not sure that we need this documentation as a copy of previous

Reply:
OK